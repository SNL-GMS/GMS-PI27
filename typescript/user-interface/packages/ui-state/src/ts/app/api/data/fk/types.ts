/**
 * Input type for Compute FK service call. This input
 * is compatible with the COI input i.e. start/end are strings
 */
import type {
  ChannelTypes,
  FkTypes,
  ProcessingMaskDefinitionTypes,
  StationTypes,
  WaveformTypes
} from '@gms/common-model';

import type { UiChannelSegment } from '../../../../types';
import type { AsyncFetchHistory } from '../../../query';

/**
 * Defines the history record type for the getChannelSegmentsByChannel query
 */
export type ComputeFkSpectraHistory = AsyncFetchHistory<FkTypes.FkInputWithConfiguration>;

/**
 * Arguments required for the ComputeFkSpectra operation
 */
export interface ComputeFkSpectraArgs {
  readonly fkSpectraDefinition: FkTypes.FkSpectraDefinition;
  readonly station: StationTypes.Station;
  readonly inputChannels: ChannelTypes.Channel[];
  readonly detectionTime: number;
  readonly startTime: number;
  readonly endTime: number;
  readonly uiChannelSegments: UiChannelSegment<WaveformTypes.Waveform>[];
  readonly processingMasksByChannel: FkTypes.ProcessingMasksByChannel[];
  readonly maskTaperDefinition: ProcessingMaskDefinitionTypes.TaperDefinition | undefined;
  readonly expandedTimeBufferSeconds: number;
}
