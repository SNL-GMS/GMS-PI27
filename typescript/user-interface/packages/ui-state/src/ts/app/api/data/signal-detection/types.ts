import type { SignalDetectionTypes } from '@gms/common-model';

/**
 * Arguments to modify existing Signal Detection
 */
export interface UpdateSignalDetectionArrivalTimeArgs {
  readonly signalDetectionId: SignalDetectionTypes.SignalDetectionIdString;
  readonly arrivalTime: {
    readonly value: number;
    readonly uncertainty: number;
  };
}

/**
 * Arguments to delete a signal detection
 */
export interface DeleteSignalDetectionArgs {
  readonly signalDetectionIds: string[];
}
