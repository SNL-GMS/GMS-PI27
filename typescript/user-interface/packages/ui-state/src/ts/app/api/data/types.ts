import type { BeamformingTemplateTypes, FkTypes, RotationTypes } from '@gms/common-model';

import type {
  AssociationConflictRecord,
  ChannelRecord,
  EventsRecord,
  FilterDefinitionsForSignalDetectionsRecord,
  FilterDefinitionsRecord,
  FkChannelSegmentRecord,
  FkFrequencyThumbnailRecord,
  QcSegmentRecord,
  SignalDetectionsRecord,
  UIChannelSegmentRecord
} from '../../../types';
import type { GetChannelsByNamesHistory } from './channel/types';
import type {
  PredictFeaturesForEventLocation,
  PredictFeaturesForEventLocationHistory
} from './event/predict-features-for-event-location';
import type {
  FindEventsByAssociatedSignalDetectionHypothesesHistory,
  GetEventsWithDetectionsAndSegmentsByTimeHistory
} from './event/types';
import type { ComputeFkSpectraHistory } from './fk/types';
import type { GetFilterDefinitionsForSignalDetectionsHistory } from './signal-detection/get-filter-definitions-for-signal-detections';
import type { GetSignalDetectionsWithSegmentsByStationAndTimeHistory } from './signal-detection/get-signal-detections-segments-by-station-time';
import type {
  FilterDefinitionByUsage,
  FilterDefinitionByUsageRecord,
  GetBeamformingTemplatesHistory,
  GetFilterDefinitionByUsageHistory,
  GetFkReviewablePhasesHistory,
  GetFkSpectraTemplatesHistory,
  GetProcessingMaskDefinitionsHistory,
  GetRotationTemplatesHistory,
  ProcessingMaskDefinitionByProcessingOperationByPhaseByChannel
} from './signal-enhancement';
import type {
  FindEventBeamsByEventHypothesisAndStationsHistory,
  UiChannelSegmentByEventHypothesisId
} from './waveform';
import type { FindQCSegmentsByChannelAndTimeRangeHistory } from './waveform/find-qc-segments-by-channel-and-time-range';
import type { GetChannelSegmentsByChannelHistory } from './waveform/get-channel-segments-by-channel';

/**
 * Defines the Data slice state.
 */
export interface DataState {
  /** the channel segments - by unique channel name - populated by multiple queries */
  uiChannelSegments: UIChannelSegmentRecord;

  /** FK channel segment records */
  fkChannelSegments: FkChannelSegmentRecord;

  /** Fk frequency thumbnails record */
  fkFrequencyThumbnails: FkFrequencyThumbnailRecord;

  /** the channels by effectiveTime populated by the getChannelsByNamesTimeRange query */
  channels: {
    raw: ChannelRecord;
    beamed: ChannelRecord;
    filtered: ChannelRecord;
  };
  /** the signal detections - by signal detection id - populated by multiple queries */
  signalDetections: SignalDetectionsRecord;
  /** the events - by time- populated by multiple queries */
  events: EventsRecord;
  eventBeams: UiChannelSegmentByEventHypothesisId;
  /** the association conflicts for events in conflict */
  associationConflict: AssociationConflictRecord;
  /** designed filter definitions */
  filterDefinitions: FilterDefinitionsRecord;
  /** filter definitions for signal detections */
  filterDefinitionsForSignalDetections: FilterDefinitionsForSignalDetectionsRecord;
  /** default filter definitions */
  defaultFilterDefinitionsMap: FilterDefinitionByUsage;
  /** qc segments by unique channel name */
  qcSegments: QcSegmentRecord;
  /** Processing Mask definitions */
  processingMaskDefinitionsByChannels: ProcessingMaskDefinitionByProcessingOperationByPhaseByChannel[];
  /** Beamforming templates */
  beamformingTemplates: BeamformingTemplateTypes.BeamformingTemplatesByBeamTypeByStationByPhase;
  /** Rotation templates */
  rotationTemplates: RotationTypes.RotationTemplateByPhaseByStationRecord;
  /** filter definitions by usage map */
  filterDefinitionsForUsage: FilterDefinitionByUsageRecord;
  /** feature predictions for event locations */
  predictFeaturesForEventLocation: PredictFeaturesForEventLocation;

  /**
   * FK Reviewable phases, organized as a record of records.
   * Top-level key is an activity name, inner record key is a station name.
   */
  fkReviewablePhases: FkTypes.FkReviewablePhasesByActivityNameByStation;
  /** FK Spectra Templates */
  fkSpectraTemplates: FkTypes.FkSpectraTemplatesByStationByPhase;
  /** query history */
  queries: {
    /** the history record of the computeFkSpectra call */
    computeFkSpectra: ComputeFkSpectraHistory;
    /** the history record of the getChannelSegmentsByChannel query */
    getChannelSegmentsByChannel: GetChannelSegmentsByChannelHistory;
    /** the history record of the findQCSegmentsByChannelAndTimeRange query */
    findQCSegmentsByChannelAndTimeRange: FindQCSegmentsByChannelAndTimeRangeHistory;
    /** the history record of the findEventBeamsByEventHypothesisAndStations query */
    findEventBeamsByEventHypothesisAndStations: FindEventBeamsByEventHypothesisAndStationsHistory;
    /** the history record of the getSignalDetectionWithSegmentsByStationAndTime query */
    getSignalDetectionWithSegmentsByStationAndTime: GetSignalDetectionsWithSegmentsByStationAndTimeHistory;
    /** the history record of the getEventsWithDetectionsAndSegmentsByTime query */
    getEventsWithDetectionsAndSegmentsByTime: GetEventsWithDetectionsAndSegmentsByTimeHistory;
    /** the history record of the findEventsByAssociatedSignalDetectionHypotheses query */
    findEventsByAssociatedSignalDetectionHypotheses: FindEventsByAssociatedSignalDetectionHypothesesHistory;
    /** the history record of the getChannelsByName query */
    getChannelsByNamesTimeRange: GetChannelsByNamesHistory;
    /** the history record of the getFilterDefinitionsByUsage query */
    getFilterDefinitionsForSignalDetections: GetFilterDefinitionsForSignalDetectionsHistory;
    /** the history record of the getProcessingMaskDefinitions query */
    getProcessingMaskDefinitions: GetProcessingMaskDefinitionsHistory;
    /** the history record of the getBeamformingTemplates query */
    getBeamformingTemplates: GetBeamformingTemplatesHistory;
    /** the history record of the getRotationTemplates query */
    getRotationTemplates: GetRotationTemplatesHistory;
    /** the history record of the getFkReviewablePhases query */
    getFkReviewablePhases: GetFkReviewablePhasesHistory;
    /** the history record of the getFkSpectraTemplates query */
    getFkSpectraTemplates: GetFkSpectraTemplatesHistory;
    /** the history record of the getFilterDefinitionsByUsageMap */
    getFilterDefinitionsByUsageMap: GetFilterDefinitionByUsageHistory;
    /** the history record of the {@link PredictFeaturesForEventLocationHistory} */
    predictFeaturesForEventLocation: PredictFeaturesForEventLocationHistory;
  };
}

export type DataStateKeys = keyof DataState;
