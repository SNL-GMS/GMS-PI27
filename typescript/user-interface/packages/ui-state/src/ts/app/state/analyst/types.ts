import type { ChannelSegmentTypes, WaveformTypes } from '@gms/common-model';
import { createEnumTypeGuard } from '@gms/common-util';
import Immutable from 'immutable';

export enum AnalystKeyAction {
  ESCAPE = 'Escape',
  SAVE_OPEN_EVENT = 'Save Open Event',
  SAVE_ALL_EVENTS = 'Save All Events in Interval'
}

export const AnalystKeyActions: Immutable.Map<string, AnalystKeyAction> = Immutable.Map([
  ['Control+KeyS', AnalystKeyAction.SAVE_OPEN_EVENT],
  ['Control+Shift+KeyS', AnalystKeyAction.SAVE_ALL_EVENTS],
  ['Escape', AnalystKeyAction.ESCAPE]
]);

/**
 * The display mode options for the waveform display.
 */
export enum WaveformDisplayMode {
  DEFAULT = 'Default',
  MEASUREMENT = 'Measurement'
}
/**
 * Available waveform align types.
 */
export enum AlignWaveformsOn {
  TIME = 'Time',
  PREDICTED_PHASE = 'Predicted',
  OBSERVED_PHASE = 'Observed'
}

/**
 * Available waveform sort types.
 */
export enum WaveformSortType {
  distance = 'Distance',
  stationNameAZ = 'Station: A-Z',
  stationNameZA = 'Station: Z-A'
}

export type DisplayedMagnitudeTypes = Record<string, boolean>;

export const isAnalystKeyAction = createEnumTypeGuard(AnalystKeyAction);

/**
 * Measurement mode state.
 */
export interface MeasurementMode {
  /** The display mode */
  mode: WaveformDisplayMode;

  /**
   * Measurement entries that are manually added or hidden by the user.
   * The key is the signal detection id
   */
  entries: Record<string, boolean>;
}

/**
 * A list such that at each index, it indicates whether the filter at the
 * same index is within the hotkey cycle.
 */
export type HotkeyCycleList = (boolean | null)[];

/**
 * The location solution state.
 * Includes:
 *   * The selected location solution set and solution
 *   * The selected preferred location solution set and solution
 */
export interface LocationSolutionState {
  selectedLocationSolutionId: string;
  selectedLocationSolutionSetId: string;
  selectedPreferredLocationSolutionId: string;
  selectedPreferredLocationSolutionSetId: string;
}

export interface LoadingActionStatus {
  isComplete?: boolean;
  error?: string;
  actionType: 'REQUEST' | 'CLIENT_SIDE_ACTION';
}

export interface RequestStatus extends LoadingActionStatus {
  url?: string;
  actionType: 'REQUEST';
}

export interface ClientActionStatus extends LoadingActionStatus {
  clientAction?: string;
  actionType: 'CLIENT_SIDE_ACTION';
}

export const isRequestStatus = (statusObj: LoadingActionStatus): statusObj is RequestStatus => {
  return statusObj?.actionType === 'REQUEST';
};

export const isClientActionStatus = (
  statusObj: LoadingActionStatus
): statusObj is ClientActionStatus => {
  return statusObj.actionType === 'CLIENT_SIDE_ACTION';
};

/**
 * Type used for determining the analyst action on an event
 */
export type EventActionTypes = 'duplicate' | 'reject' | 'delete' | 'open' | 'close' | 'details';

/**
 * Type used for determining the analyst action on a signal detection
 */
export type SignalDetectionActionTypes =
  | 'reject associate'
  | 'delete'
  | 'details'
  | 'associate'
  | 'unassociate'
  | 'phase'
  | 'default phase'
  | 'current phase'
  | 'fk'
  | 'rotate';

/**
 * union of action types
 */
export type ActionTypes = EventActionTypes | SignalDetectionActionTypes | null;

export interface ActionTarget {
  actionType: ActionTypes; // the type of action the analyst will take on an event or sd
  eventIds: string[]; // The ids of the events that are the target of a user's action
  signalDetectionIds: string[];
}

export interface AnalystState {
  channelFilters: Record<string, WaveformTypes.WaveformFilter>;
  defaultSignalDetectionPhase: string;
  effectiveNowTime: number;
  fileId: {
    workflowUniqueId: string;
    fileHandle: FileSystemFileHandle | null;
  };
  hotkeyCycleOverrides: Record<string, Record<number, boolean>>;
  location: LocationSolutionState;
  measurementMode: MeasurementMode;
  requestTracker: {
    /* record keyed on ids */
    requests: Record<string, RequestStatus | ClientActionStatus>;
    initiatedRequests: number;
    completedRequests: number;
    lastRequestId: string;
  };
  openEventId: string;
  actionTargets: ActionTarget;
  currentPhase: string;
  selectedEventIds: string[];
  selectedFilterIndex: number | null;
  selectedFilterList: string;
  selectedSdIds: string[];
  selectedSortType: WaveformSortType;

  eventListOpenEventTriggered: boolean;
  mapOpenEventTriggered: boolean;
  alignWaveformsOn: AlignWaveformsOn;
  phaseToAlignOn: string;
  phaseSelectorPhaseList: string;
  phaseSelectorFavorites: Record<string, string[]>;
  selectedWaveforms: ChannelSegmentTypes.ChannelSegmentDescriptor[];
}
