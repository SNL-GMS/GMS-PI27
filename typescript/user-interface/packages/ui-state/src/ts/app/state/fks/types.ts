/**
 * Different filters that are available
 */

import type { FkTypes } from '@gms/common-model';

import type { ChannelFilterRecord } from '../../../types';

export enum FkThumbnailsFilterType {
  KEYACTIVITYPHASES = 'Key Activity Phases',
  ALL = 'All',
  OPENEVENT = 'Open Event',
  NEEDSREVIEW = 'Needs review'
}

export interface FksState {
  sdIdsToShowFk: string[];
  displayedSDMeasuredValues: FkTypes.FkMeasuredValues | undefined;
  fkPlotsExpandToolbar: boolean;
  currentFkThumbnailFilter: FkThumbnailsFilterType;
  fkChannelFilters: Record<string, ChannelFilterRecord>;
  displayedSignalDetectionId: string;
}
