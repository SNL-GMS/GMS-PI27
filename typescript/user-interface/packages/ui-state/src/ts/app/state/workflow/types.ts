import type { CommonTypes, WorkflowTypes } from '@gms/common-model';
import type { Nullable } from '@gms/common-model/lib/type-util/type-util';

export interface WorkflowState {
  timeRange: Nullable<CommonTypes.TimeRange>;
  stationGroup: WorkflowTypes.StationGroup;
  openIntervalName: string; // e.x AL1
  openActivityNames: string[]; // e.x Event Review
  analysisMode: WorkflowTypes.AnalysisMode | null;
}
