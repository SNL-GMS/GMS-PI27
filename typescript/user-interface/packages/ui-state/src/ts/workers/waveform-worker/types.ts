import type { ChannelTypes, StationTypes, WaveformTypes } from '@gms/common-model';
import type { BeamDefinition } from '@gms/common-model/lib/beamforming-templates/types';
import type { ProcessingMask, TimeRangesByChannel } from '@gms/common-model/lib/channel-segment';
import type { TimeRange } from '@gms/common-model/lib/common';
import type { AxiosRequestConfig } from 'axios';

import type { GetChannelSegmentsByChannelQueryArgs } from '../../app/api/data/waveform/get-channel-segments-by-channel';
import type { UiChannelSegment } from '../../types';

/**
 * The colors to use when building the weavess channel segments
 */
export interface ChannelSegmentColorOptions {
  waveformColor: string;
  labelTextColor: string;
}

export interface AmplitudeBounds {
  amplitudeMax: number;
  amplitudeMin: number;
  amplitudeTotal?: number;
  totalSamplesCount?: number;
  amplitudeMaxSecs?: number;
  amplitudeMinSecs?: number;
}

/**
 * An axios request known to have a waveform query request in its data.
 */
export type WaveformAxiosRequestConfig = AxiosRequestConfig & {
  data: GetChannelSegmentsByChannelQueryArgs;
};

// Props for the ui-beam-processor maskAndBeamWaveforms operation
export interface MaskAndBeamWaveformWorkerProps {
  beamDefinition: BeamDefinition;
  beamStartTime: number;
  beamEndTime: number;
  station: StationTypes.Station;
  filteredChannelSegments: UiChannelSegment<WaveformTypes.Waveform>[];
  processingMasksByChannel: {
    channel: ChannelTypes.Channel;
    processingMasks: ProcessingMask[];
  }[];
  currentInterval: TimeRange;
  missingInputChannels: TimeRangesByChannel[];
}

export interface MaskAndBeamWaveformResult {
  channel: ChannelTypes.Channel;
  uiChannelSegment: UiChannelSegment<WaveformTypes.Waveform>;
}
