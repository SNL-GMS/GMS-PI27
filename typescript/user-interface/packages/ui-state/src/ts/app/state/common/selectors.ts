import type { AppState } from '../../store';
import type { GLDisplayState } from './types';

export const isKeyboardShortcutPopupOpen = (state: AppState): boolean =>
  state.app.common.keyboardShortcutsVisibility;

export const isUserPreferencesPopupOpen = (state: AppState): boolean =>
  state.app.common.userPreferencesVisibility;

export const isCommandPaletteOpen = (state: AppState): boolean =>
  state.app.common.commandPaletteIsVisible;

/**
 * Redux selector that returns the selected station and channel ids
 */
export const selectSelectedStationsAndChannelIds = (state: AppState): string[] =>
  state.app.common.selectedStationIds;

/**
 * Redux selector that returns the open golden layout displays
 */
export const selectOpenGoldenLayoutDisplays = (state: AppState): Record<string, GLDisplayState> =>
  state.app.common.glLayoutState;
