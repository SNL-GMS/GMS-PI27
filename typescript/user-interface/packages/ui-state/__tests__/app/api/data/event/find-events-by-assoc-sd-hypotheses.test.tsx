import { CommonTypes, SignalDetectionTypes } from '@gms/common-model';
import { Units } from '@gms/common-model/lib/common/types';
import type { AnyAction } from 'redux';
import type { MockStoreCreator } from 'redux-mock-store';
import createMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import type { AppState } from '../../../../../src/ts/app';
import {
  addFindEventsByAssociatedSignalDetectionHypothesesMatchReducers,
  findEventsByAssociatedSignalDetectionHypotheses,
  findEventsByAssociatedSignalDetectionHypothesesQuery
} from '../../../../../src/ts/app/api/data/event/find-events-by-assoc-sd-hypotheses';
import type { FindEventsByAssociatedSignalDetectionHypothesesQueryArgs } from '../../../../../src/ts/app/api/data/event/types';
import { appState } from '../../../../test-util';

const mockArgs: FindEventsByAssociatedSignalDetectionHypothesesQueryArgs = {
  signalDetectionHypotheses: [
    {
      id: {
        id: '20cc9505-efe3-3068-b7d5-59196f37992c',
        signalDetectionId: '012de1b9-8ae3-3fd4-800d-58665c3152cc'
      },
      parentSignalDetectionHypothesis: null,
      deleted: false,
      station: {
        name: 'ASAR',
        effectiveAt: 0
      },
      monitoringOrganization: 'GMS',
      featureMeasurements: [
        {
          channel: {
            name: 'ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4',
            effectiveAt: 1636503404
          },
          measuredChannelSegment: {
            id: {
              channel: {
                name: 'ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4',
                effectiveAt: 1636503404
              },
              startTime: 1636503404,
              endTime: 1636503704,
              creationTime: 1636503404
            }
          },
          measurementValue: {
            arrivalTime: {
              value: 1636503404,
              standardDeviation: 1.162,
              units: CommonTypes.Units.SECONDS
            },
            travelTime: { value: 0, units: Units.COUNTS_PER_NANOMETER }
          },
          snr: {
            value: 8.9939442,
            standardDeviation: 0,
            units: CommonTypes.Units.DECIBELS
          },
          featureMeasurementType: SignalDetectionTypes.FeatureMeasurementType.ARRIVAL_TIME
        }
      ]
    }
  ],
  stageId: { name: 'AL1' }
};

describe('Find events by associated SD hypotheses', () => {
  test('contains definitions', () => {
    expect(addFindEventsByAssociatedSignalDetectionHypothesesMatchReducers).toBeDefined();
    expect(findEventsByAssociatedSignalDetectionHypotheses).toBeDefined();
    expect(findEventsByAssociatedSignalDetectionHypothesesQuery).toBeDefined();
  });

  test('query is skipped when args are falsy, not skipped when args are truthy', () => {
    const mockSkipArgs: FindEventsByAssociatedSignalDetectionHypothesesQueryArgs = {
      signalDetectionHypotheses: [],
      stageId: { name: 'test' }
    };
    expect(
      findEventsByAssociatedSignalDetectionHypothesesQuery.shouldSkip(mockSkipArgs)
    ).toBeTruthy();
    expect(findEventsByAssociatedSignalDetectionHypothesesQuery.shouldSkip(mockArgs)).toBeFalsy();
  });

  it('will not execute query if the args are invalid', async () => {
    const mockStoreCreator: MockStoreCreator<AppState, AnyAction> = createMockStore([thunk]);

    const store = mockStoreCreator(appState);

    await store.dispatch(
      findEventsByAssociatedSignalDetectionHypotheses({
        ...mockArgs,
        signalDetectionHypotheses: []
      }) as any
    );

    // results should have empty arrays since current interval is not set
    expect(store.getActions()).toHaveLength(0);
  });

  test('findEventsByAssociatedSignalDetectionHypotheses returns events', () => {
    expect(findEventsByAssociatedSignalDetectionHypotheses(mockArgs)).toMatchSnapshot();
  });

  test('addFindEventsByAssociatedSignalDetectionHypothesesMatchReducers builds', () => {
    const builderMap: any[] = [];
    const builder: any = {
      addCase: (k, v) => {
        builderMap.push(v);
        return builder;
      },
      addMatcher: (k, v) => {
        builderMap.push(v);
        return builder;
      }
    };

    addFindEventsByAssociatedSignalDetectionHypothesesMatchReducers(builder);
    expect(builderMap).toMatchSnapshot();
    const state = { queries: { findEventsByAssociatedSignalDetectionHypotheses: {} } };
    // eslint-disable-next-line prefer-const
    let action = {
      meta: {
        requestId: findEventsByAssociatedSignalDetectionHypothesesQuery.idGenerator(mockArgs),
        mockArgs
      },
      payload: []
    };

    builderMap[0](state, action);
    expect(state).toMatchSnapshot();
    builderMap[1](state, action);
    expect(state).toMatchSnapshot();
    builderMap[2](state, action);
    expect(state).toMatchSnapshot();
  });
});
