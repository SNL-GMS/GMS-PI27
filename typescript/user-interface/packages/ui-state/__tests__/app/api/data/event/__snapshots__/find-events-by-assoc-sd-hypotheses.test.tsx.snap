// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find events by associated SD hypotheses addFindEventsByAssociatedSignalDetectionHypothesesMatchReducers builds 1`] = `
[
  [Function],
  [Function],
  [Function],
]
`;

exports[`Find events by associated SD hypotheses addFindEventsByAssociatedSignalDetectionHypothesesMatchReducers builds 2`] = `
{
  "queries": {
    "findEventsByAssociatedSignalDetectionHypotheses": {
      "stage:AL1-sdIds:20cc9505-efe3-3068-b7d5-59196f37992c": {
        "stage:AL1-sdIds:20cc9505-efe3-3068-b7d5-59196f37992c": {
          "arg": undefined,
          "attempts": undefined,
          "error": {},
          "status": "pending",
          "time": undefined,
        },
      },
    },
  },
}
`;

exports[`Find events by associated SD hypotheses addFindEventsByAssociatedSignalDetectionHypothesesMatchReducers builds 3`] = `
{
  "queries": {
    "findEventsByAssociatedSignalDetectionHypotheses": {
      "stage:AL1-sdIds:20cc9505-efe3-3068-b7d5-59196f37992c": {
        "stage:AL1-sdIds:20cc9505-efe3-3068-b7d5-59196f37992c": {
          "arg": undefined,
          "attempts": 1,
          "error": {},
          "status": "fulfilled",
          "time": 0.25,
        },
      },
    },
  },
}
`;

exports[`Find events by associated SD hypotheses addFindEventsByAssociatedSignalDetectionHypothesesMatchReducers builds 4`] = `
{
  "queries": {
    "findEventsByAssociatedSignalDetectionHypotheses": {
      "stage:AL1-sdIds:20cc9505-efe3-3068-b7d5-59196f37992c": {
        "stage:AL1-sdIds:20cc9505-efe3-3068-b7d5-59196f37992c": {
          "arg": undefined,
          "attempts": 2,
          "error": {},
          "status": "rejected",
          "time": 0.25,
        },
      },
    },
  },
}
`;

exports[`Find events by associated SD hypotheses findEventsByAssociatedSignalDetectionHypotheses returns events 1`] = `[Function]`;
